OpenSim/Common/Exception.h

```diff
@@ -47,7 +47,31 @@
 //=============================================================================
 //=============================================================================
 
-namespace OpenSim { 
+namespace OpenSim {
+
+
+/** Macros to throw OpenSim exceptions. The purpose of these macros is to avoid
+having to provide the first few arguments (which are common to all OpenSim
+exceptions) to the exception constructor in the throw statements. This also
+allows us to add more details (eg class name) later easily.                   
+
+Note -- Extra braces enclosing "if" are to avoid problems when these macros are
+called within if-else statements like:
+\code{.cpp}
```

Comment:
The macros are undocumented: http://myosin.sourceforge.net/628/

I think you might have to modify doxyfiles to get the macros to show up in doxygen. Also if you want both macros to share the same doxygen comment, you could maybe use a @group? I'm not sure, but could you make sure the documentation for these macros shows up in doxygen in a nice way? I can help you play around with the doxyfiles if necessary.